test/test_railwaynetwork.cpp

```diff
@@ -1,1085 +1,2469 @@
+#include "Definitions.hpp"
```

Comment:
Since you do many input sanity checks in public methods, it might be good to also check that these exceptions are actually thrown. This also increases coverage of course.

Reply:
I would propose increasing coverage in a new PR